index.js

```diff
@@ -239,98 +240,120 @@ module.exports = {
     return {
       post: {
         // Route to accept the submitted form.
-        submit: async function (req) {
-          const input = req.body;
-          const output = {};
-          const formErrors = [];
-          const formId = self.inferIdLocaleAndMode(req, req.body._id);
-
-          const form = await self.find(req, {
-            _id: self.apos.launder.id(formId)
-          }).toObject();
-
-          if (!form) {
-            throw self.apos.error('notfound');
-          }
+        submit: [
+          require('connect-multiparty')(),
+          async function (req) {
+            const input = JSON.parse(req.body.data);
+            const output = {};
+            const formErrors = [];
+            const formId = self.inferIdLocaleAndMode(req, input._id);
+
+            const form = await self.find(req, {
+              _id: self.apos.launder.id(formId)
+            }).toObject();
+
+            if (!form) {
+              throw self.apos.error('notfound');
+            }
 
-          if (form.enableRecaptcha) {
-            try {
-              // Process reCAPTCHA input if needed.
-              await self.checkRecaptcha(req, input, formErrors);
-            } catch (e) {
-              console.error('reCAPTCHA submission error', e);
-              throw self.apos.error('invalid');
+            if (form.enableRecaptcha) {
+              try {
+                // Process reCAPTCHA input if needed.
+                await self.checkRecaptcha(req, input, formErrors);
+              } catch (e) {
+                self.apos.util.error('reCAPTCHA submission error', e);
+                throw self.apos.error('invalid');
+              }
+            }
+
+            // Find any file field submissions and insert the files as attachments
+            for (const [ field, value ] of Object.entries(input)) {
```

Comment:
Could the form fields be nested somehow or is that not a thing with forms?

Reply:
That's not a thing with actual input fields. We have the conditional widget that uses nesting.

Reply:
Yeah, these "fields" are actual form input values, so the conditional widget isn't a factor.